### DISCUSSION TITLE
Is there any way to pass the result of one async method as a parameter to another async method?

### DISCUSSION THREAD

**Discussion by dipali2525** [vkhorikov/CSharpFunctionalExtensions#380] [Q&A]
Is there any way to pass the result of one async method as a parameter to another async method?
I want method chaining.

For example: -
```
return  await this.CreateMembers(models)
                      .Check(t => this.CreateLogins(t))
                      .Check(t => this.CreateMemberMenus(t))
                     .Check(t => this.CreateAuditTrail(t));
```

**Comment by vkhorikov** [comment]
There should. If you can't, it's most likely because there's an overload missing. You can check how other methods that fully support async overloads (e.g `Map`) are done and implement this one similarly. PRs are welcome.
